### ISSUE TITLE
Is it possible to avoid getting to close to a non walkable cell ?

### ISSUE THREAD

**Issue by awddawawd** [brean/python-pathfinding#47]
Hi,
I'm trying to create a path finding method for the game brawlstars and I was wondering if it could be possible to find a path that does take a little longer route but avoid touch a neighbor cell that is non walkable. This would be helpful for my project since in brawlstars, there's some kind of "auto adjustment" when you go straight to a wall that makes the player move a little in order to avoid getting hit, but this feature is quite annoying for a bot and can make it loose the track of the player. 

For example this :
![large_path](https://github.com/brean/python-pathfinding/assets/57504682/f29b0710-361e-4095-8504-dc4a97243803)

instead of the current that is :
![narrow_path](https://github.com/brean/python-pathfinding/assets/57504682/059ee43c-158b-42ea-9851-e51c8820f63f)

**Comment by brean** [comment]
Yes, you can use the already implemented weighted-grid feature of python-patfinding.
You just need to change your map to get higher cost values around your obstacles with increasing proximity.
I don't have an example for this, also there is no implementation of it yet, but you could probably copy the concept of costmaps from ROS: http://wiki.ros.org/costmap_2d/hydro/inflation
I could help you with an implementation if you have problems but I encourage you to try yourself first. 
Feel free to provide a PR, ideally with a bigger map for testing, a costmap generation algorithm, unit-tests and documentation in markdown for python-pathfinding.

**Comment by brean** [comment]
Also extending python-pathfinding with a Theta*-implementation would be awesome 
(see https://news.movel.ai/theta-star )
